File: src/main/scala/ch/datascience/typesystem/DataType.scala
package ch.datascience.typesystem

/**
  * Created by johann on 07/03/17.
  */
sealed abstract class DataType

object DataType {

  case object String extends DataType
  case object Character extends DataType
  case object Boolean extends DataType
  case object Byte extends DataType
  case object Short extends DataType
  case object Integer extends DataType
  case object Long extends DataType
  case object Float extends DataType
  case object Double extends DataType
  case object Decimal extends DataType
  case object Precision extends DataType
  case object Date extends DataType
  case object Geoshape extends DataType
  case object UUID extends DataType

}

File: src/main/java/ch/datascience/typesystem/model/DataType.java
package ch.datascience.typesystem.model;

/**
 * Created by johann on 17/03/17.
 */
public enum DataType {

    STRING,
    CHARACTER,
    BOOLEAN,
    BYTE,
    SHORT,
    INTEGER,
    LONG,
    FLOAT,
    DOUBLE;
//    Decimal
//    Precision
//    Date
//    Geoshape
//    UUID

    public Class<?> javaClass() {
        switch (this) {
            case STRING:
                return java.lang.String.class;
            case CHARACTER:
                return java.lang.Character.class;
            case BOOLEAN:
                return java.lang.Boolean.class;
            case BYTE:
                return java.lang.Byte.class;
            case SHORT:
                return java.lang.Short.class;
            case INTEGER:
                return java.lang.Integer.class;
            case LONG:
                return java.lang.Long.class;
            case FLOAT:
                return java.lang.Float.class;
            case DOUBLE:
                return java.lang.Double.class;
            default:
                throw new RuntimeException("No java class mapping for: " + this.name());
        }
    }

}
